### DISCUSSION TITLE
Trying to understand 50 Doge fees on 1 Doge with default 0.01 fees

### DISCUSSION THREAD

**Discussion by qlpqlp** [dogecoin/dogecoin#2747] [Help needed] [answered]
Hello to all :)

Im confused about this that happend to me now trying to make some more testes on NFT's :)

https://blockchair.com/dogecoin/transaction/922478ac5845051c852fcb7b953eaf83f02a4d9aed51346fda2dd24d51908291

Can anyone try to explain why?

Command send
createrawtransaction([array("txid" => "922478ac5845051c852fcb7b953eaf83f02a4d9aed51346fda2dd24d51908291", "vout" => 0)],array("DTEZbVHMhseLF6qqArdqDiipgVWd7Wz6J6" => 1)));

signed and send

thanks in advance :)

**Comment by rnicoll** [comment, marked as answer]
Oof.

So, the command you've sent is basically:

1. Spend output 0 of 922478ac5845051c852fcb7b953eaf83f02a4d9aed51346fda2dd24d51908291
2. Send 1 Doge to DTEZbVHMhseLF6qqArdqDiipgVWd7Wz6J6

Notably, you've not told it what to do with the rest of the transaction output being spent in step 1, and while if you simply told it to send coins it would have sent the rest to a change address, with createrawtransaction it just presumes you know what you meant. However the fee is the difference between the input value and the output value, and so if you spend 51.09214078 DOGE and output 1 DOGE, the 50.09214078 are burnt as fee.

Does that help?

**Reply by qlpqlp** [reply]
lol , Many thanks for answer so quick :)

Objetive: Use same transaction ID to send Doge to anouther address witouth making my mistake above :)

Can this be done like this?

```
createrawtransaction([array("txid" => "922478ac5845051c852fcb7b953eaf83f02a4d9aed51346fda2dd24d51908291", "vout" => 0)],array("DTEZbVHMhseLF6qqArdqDiipgVWd7Wz6J6" => 1)));
fundrawtransaction(hex);
signrawtransaction(hex);
sendrawtransaction(hex);
```

or

like this;

`
createrawtransaction([array("txid" => "922478ac5845051c852fcb7b953eaf83f02a4d9aed51346fda2dd24d51908291", "vout" => 0)],array("DTEZbVHMhseLF6qqArdqDiipgVWd7Wz6J6" => 1, "OUTHER-DOGE-ADDRESS" => 50.09214078)));
signrawtransaction(hex);
sendrawtransaction(hex);
`
Thanks for any help :)

**Reply by qlpqlp** [reply]
Found it :)

Many thanks :)

Now doing some final tests to show how to and the fresh news about NFT's on Dogecoin :)

Hope to meet you all Dooge Core Devs some day in person eheh :)

Crazy week (thats why i said day 13 is my lucky day lol)
My initial intenttion was not NFT's but wen i was starting to test it i think to myself (i initially worked in digital documents and microfilm for storage and index documents), why not "Checksum" eheh

**Reply by rnicoll** [reply]
BTW I'd rather presumed you'd be putting the NFT hash into the transactions somewhere, although I acknowledge that's going to get into custom scripting potentially and even trickier :-/ Still... have a look at how transaction scripts work, you can actually embed data into a transaction which would give you a much saner audit trail of what's going on

**Reply by patricklodder** [reply]
Note that using anything other than OP_RETURN to create something non-spendable will create a txout that will pollute the utxo db, forever. Please use OP_RETURN, so that at least the spam won't be kept in RAM.

**Reply by qlpqlp** [reply]
Many thanks @patricklodder and @rnicoll :)

Im trying only to use this:

// this will store a Checksum of a original QR Code image With a JSON info/foto/video/music/document/zip file/Hashed fingerprint/imagination is the limit

createrawtransaction([array("txid" => $txid, "vout" => 0)],array($Such_Address => 0.01,"data" => $Such_NFT,$Change_Address => (($Total_Doge - 0.01)-$fees))));;

The "sell" / "transfer" to the new owner uses the same TX id of the NFT and sends like this:

createrawtransaction([array("txid" => $nft_txid, "vout" => 0)],array($Such_New_Owner_Address => 0.01,"data" => $Such_NFT,$Change_Address => (($Total_Doge - 0.01)-$fees))));;

Is that OK with you all? any outher ideas or better methods? :)

**Reply by qlpqlp** [reply]
> OP_RETURN

For what I read here: https://en.bitcoin.it/wiki/OP_RETURN
i think i now understand what you are talking about.

If I dont spent all Doge in the transaction to use the OP_RETURN/data field, it will be on RAM forever, polluting.
Is that correct?

So the solution is to try to spent all Doge in the transaction used to add data into OP_RETURN/data, correct?

- Wen first send the Data/OP_RETURN, spend all in that transaction.
- Then fund the transaction if needed to "transfer/sell" wen i want to send anouther Data/OP_RETURN and try to spent all Doge in it also

Is it correct?

Sorry to all this questions, you have more things to do then reply to me :)

**Reply by qlpqlp** [reply]
Can you check if this pollute the ram? 6015fe57e372fd11747e3edb4e1eec0100c7209d497b43415279c0bf1d644c6b

**Reply by patricklodder** [reply]
Your second proposal ([`6015fe57`](https://chain.so/tx/DOGE/6015fe57e372fd11747e3edb4e1eec0100c7209d497b43415279c0bf1d644c6b)) leaves no utxo footprint and only transient mempool and chain/block capacity pressure, but likely leaves you with the issue that there is no intrinsic mechanism to transfer the output, meaning that without having any overlay protocol, you basically burned it with that.

An utxo (unspent transaction output) is fine if it is spendable... By attaching an OP_RETURN to an utxo that contains DOGE, you're making the accompanying spendable utxo inherently non-fungible too. It's less than optimal, though of course effectively the least intrusive way of doing this other than doing yet another counterparty/omni implementation. I have 2 thoughts about that:

1. You need a way to inform/enforce non-fungibility of your utxo that came with your `OP_RETURN` transaction on the wallet side - i.e. Dogecoin Core's coin selection algorithm will just spend it by default if it fits a target price for a transaction, and that will make people accidentally burn whatever data they may have wanted to transfer in the future.
2. As long as your OP_RETURN isn't burned, there is pressure on the utxo database with the additional output that you're creating because the owner of that utxo has a clear disincentive to spend, unlike all the other, fungible, utxos, where there (should be) an incentive to spend.
 
I think it's important to realize that using an utxo chain is more suited toward fungibility than non-fungibility, as it was designed for that, which is why most (if not all) of the popular NFT platforms don't use utxo chains.

**Reply by qlpqlp** [reply]
Thanks for answer.

independent of anything, wen  i create a transaction spendable or not does the TX ID will for ever be stored with the same OP_RETURN data? or can be changed/removed by the owner or anyone?

**Reply by patricklodder** [reply]
The transaction hash (in combination with block height AND hash, just to bar the extremely unlikely event of a collision) will be unique. It will forever be stored as long as it got mined, the block did not get orphaned and there is someone running a full node containing the full chain.

**Reply by qlpqlp** [reply]
That was my only concern, and, so if its stored on the Dogecoin Blockchain with the Checksum of a file, so its a real NFT of an asset 🤪

Much wow!

Lot to talk with u all for the NFT platform :) to ask for consensus on all :)

**Reply by patricklodder** [reply]
I think what you have is a timestamped data on the Dogecoin blockchain. Whatever meaning, value or assets you externally assign to that is not by any means guaranteed by any Dogecoin network protocol, and you will very likely have problems to assign that asset any legal meaning by this transaction output alone - so it's perhaps a bit of a stretch to call it an NFT of a tokenized asset on Dogecoin. It's more like you're using the Dogecoin chain as a distributed database to store extremely-expensive-to-mutate data.

**Reply by qlpqlp** [reply]
Many thanks once again :)

In your point of view, is there any Crypto that guarantees the assignment of an asset or value? exemple ETH?

also, sorry many questions eheh, a Checksum of an digital asset stored on a field in a Blockchain, does not validate that the asset is original and was not changed? 

Dosent sound to you that an NFT purpose is to certificate an original asset that was not changed and it belongs to x person?

Don't be mad haha :) had to ask lol

**Reply by patricklodder** [reply]
>  a Checksum of an digital asset stored on a field in a Blockchain, does not validate that the asset is original and was not changed?

Not at all. You hack my laptop, take a picture of me in my undies you find there, hash it and put the hash on-chain. It doesn't mean you own it, it doesn't mean that you didn't change it and it definitely doesn't mean it's original at all because maybe I photoshopped it. But that was not what I was saying.

Example: what if I put the same hash in a transaction - who owns it then, the first transaction that ever mentioned it? What if I front-run you while your tx is still in the mempool - do I own your asset now? None of this is enforced by the Dogecoin network so you'll need something decide which transaction is valid and which one isn't. There are some smart contract standards that implement these kind of things. Now I guess that what you want to do is write software on top that parses the blockchain and tells you what's valid. That's cool, but then your software does the NFT part, not unlike other software that runs on top of the Dogecoin chain.

**Reply by qlpqlp** [reply]
Much thanks once again :)

I personally Disagree with some of your points, because what your are saying is indirectly nothing can be proven its yours or original, even with an external software :)

if i hash my dna or my finguer print and put  on the Dogecoin field is that not prove that is mine and not changed and is stored on a Oracle Barkley DB replicated around the world?

Im the living prove, its mine :)

Ofcourse you need a 3rd partie to analyse my blood and finguer print, but that is another story, the certificate was always there you only double checkd  :)

The idea of an NFT is that even if someone copy, hack your laptop  take a picture and hashed will always be the original and unmodified picture even if its an hacked and/or photoshop :)

My idea is that with a simple checksum we can have privacy on NFTs (to not copy outhers "first", i named an NBT (Non-banana Token)). 

A litle unic generated checksum token  from a digital asset, stored on the Dogecoin Blockchain Barkley DB that anyone can build an external DB intexed by the transaction id on the Dogecoin to be able to securely see his/her NBTs

Exemple one: an Checksum of a zip file encrypted with a password containing the private key of my Dogecoin Address's

i make my own DB that is connected to the transaction ID that contains my hashed NBT and confirm that the zip file, stored on a descentralized IPFS like Storj/Filecoin/Etc  is that one and was not modified.

Or to sell my music and i have put the NBT on the Blockchain now 2021 then 50 years from now i say i still have the original song and send to the new owner but wen is checking e sees is not the original, the original was with a different NBT.

In the end like @michilumin said, "finguer in slow mo" some people, that can be done lol

but you are also right, it has a big cost, burn + resources. 

In several tests i made, my core node only could verify a transaction after a couple of large seconds 
for 3 confirmations can take between 30 seconds or more,
for 5 confirmations can take 1 minute or more
and i think what is the solution?  direct P2P centralize verification?

for a fisical store, a supermarket its to much time, it needs to check in 5 seconds max.  

Does the NBT idea will ruine it the ideal fast transaction + lower fees?

Much wof to all of you :) 

its 03:00 am hehe lol, still awake because of my litle girl that wake me up lol, teying to put her to sleep again :)

**Reply by patricklodder** [reply]
I'm glad that you agree that ownership or even correctness of things outside the blockchain are very hard to prove on a blockchain. After all, there are people that have been working on trying to solve that for a long time. I hope you're not dismissing their work.

I think it's a pity that you didn't address the weakness I highlighted, regardless of what you're trying to market here. I'm also not sure what you're saying with regards to transaction confirmation speeds or how your solution solves that too.

**Reply by qlpqlp** [reply]
Much Oof :)
Sometimes i have to read 3, 4 times or more to understand 🤣
Yes, i think i know what you mean about the weakness :) Nothing in 💯 is digital world

I didnt search for any outher people that where working on it because many times the language used to identify the functionality i wanted to achieve are different, exemple Doge Party XDP, never heard of it until they started to spam averywhere lol

One thing i really like is the shortcut, an easy, simple, clean way to do things. If we complicate, for many old devs like me its a nightmare to try to learn new methods :)

Exemple: Ruby, nodeJS, Angular etc, they help on complicated projects, but for the simple ones, it dosent make sense.

So, i see a way, that i think it can work, i never give up trying to find a way to really work even if its not perfect, someone will grab and do it better :)

Love to share touhts, love to read codsinf and more to do try/error lol, debug is my thing haha!

Will share more things and ask opinions this weekend. Will code almost all day. Im already excited again for do something for Doge :) Dozens of ideas on my head 🤪

Have a wonderful day of coding to all Dogecoin Core Devs ❤️❤️❤️🐶🐶🐶
